ui/app/src/Views/SilenceForm/Updates.elm

```diff
@@ -115,72 +169,84 @@ updateForm msg form =
                 { form | matchers = matchers }
 
 
-update : SilenceFormMsg -> Model -> ( Model, Cmd Msg )
+update : SilenceFormMsg -> Model -> ( Model, Cmd SilenceFormMsg )
 update msg model =
     case msg of
-        CreateSilence silence ->
-            ( model
-            , Silences.Api.create silence
-                |> Cmd.map (SilenceCreate >> MsgForSilenceForm)
-            )
+        CreateSilence ->
+            case toSilence model.form of
+                Just silence ->
+                    ( { model | silenceId = Loading }
+                    , Silences.Api.create silence |> Cmd.map SilenceCreate
+                    )
+
+                Nothing ->
+                    ( { model
+                        | silenceId = Failure "Could not submit the form, Silence is not yet valid."
+                        , form = validateForm model.form
+                      }
+                    , Cmd.none
+                    )
 
-        SilenceCreate silence ->
-            case silence of
-                Success id ->
-                    ( model, Navigation.newUrl ("/#/silences/" ++ id) )
+        SilenceCreate silenceId ->
```

Comment:
We can definitely re-think these names I came up with. `SilenceCreate` and `CreateSilence` could definitely be better.

(note for the future)